server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -179,7 +180,9 @@
 
     private static final String SNAPSHOT_INDEX_CODEC = "snapshots";
 
-    private static final String DATA_BLOB_PREFIX = "__";
+    private static final String UPLOADED_DATA_BLOB_PREFIX = "__";
+
+    private static final String VIRTUAL_DATA_BLOB_PREFIX = "v__";
```

Comment:
Still need to have virtual blob here so that the logic/assertions in `BlobStoreIndexShardSnapshot(s)` works without format change and things stay BwC. Technically we could do without this kind of "ID", but then we'd have to go through the whole dance of only updating the format and not uploading the blob once all the snapshots are newer than version X. If we do it this way, we get the benefit of faster restores and snapshots right away since the meta hash works the same way in `6.x` already hence even in `7.7` all possible restores should work out.